pkg/valid/validations.go

```diff
@@ -22,18 +22,19 @@ import (
 )
 
 const (
-	MaxAppNameLen  = 39
-	AppNameRegExp  = `\A[a-z0-9]+(?:-[a-z0-9]+)*\z`
-	TeamNameRegExp = AppNameRegExp
-	EnvNameRegExp  = AppNameRegExp
-	CommitIDRegExp = `^[0-9a-fA-F]{40}$`
+	MaxAppNameLen        = 39
+	AppNameRegExp        = `\A[a-z0-9]+(?:-[a-z0-9]+)*\z`
+	TeamNameRegExp       = AppNameRegExp
+	EnvNameRegExp        = AppNameRegExp
+	SHA1CommitIDLength   = 40
+	commitIDPrefixRegExp = `^[0-9a-fA-F]*$`
```

Comment:
Perhaps this is a better way for writing this regex.

```suggestion
	commitIDPrefixRegExp = `^[0-9a-fA-F]{1,40}$`
```

Reply:
I think it's better to check the length separately, as this gives you the ability to generate more precise error messages.